listener.go

```diff
@@ -46,13 +46,33 @@ func ListenerWithConnectionString(connStr string) ListenerManagementOption {
 	}
 }
 
-// ListenerWithManagedIdentity configures a listener with the attached managed identity and the Service bus resource name
+// Deprecated use ListenerWithManagedIdentityClientID instead
```

Comment:
deprecate unspecific listener creator, add ClientID and ResourceID funcs instead